sql/expression/aggregation.go

```diff
@@ -158,3 +159,109 @@ func (m *Min) Merge(buffer, partial sql.Row) error {
 func (m *Min) Eval(buffer sql.Row) (interface{}, error) {
 	return buffer[0], nil
 }
+
+// Avg node to calculate the average from numeric column
+type Avg struct {
+	UnaryExpression
+}
+
+var AverageNoNumericErr = errors.NewKind("avg aggregation can only be applied on numeric columns")
```

Comment:
Check the actual behavior of MySQL AVG function; if the type of the column is not numeric, the result is 0. Check other corner cases and impement it as MySQL AVG.

Reply:
@ajnavarro  done